utils/file_backend_local.go

```diff
@@ -40,6 +42,34 @@ func (b *LocalFileBackend) ReadFile(path string) ([]byte, *model.AppError) {
 	}
 }
 
+func (b *LocalFileBackend) CopyFile(oldPath, newPath string) *model.AppError {
+	if err := os.MkdirAll(filepath.Dir(filepath.Join(b.directory, newPath)), 0774); err != nil {
+		return model.NewAppError("copyFile", "api.file.move_file.rename.app_error", nil, err.Error(), http.StatusInternalServerError)
+	}
+
```

Comment:
We have a function that does most of this (`utils.CopyFile`). You could just put this right here:

```go
if err := CopyFile(filepath.Join(b.directory, oldPath), filepath.Join(b.directory, newPath)); err != nil {
    return model.NewAppError("moveFile", ...)
}
return nil
```

In addition to not having the same function implemented twice, you get better permission handling since the other function copies them over.

Reply:
Nice. Didn't realize that the local implementations were meant to pass through to the `utils/file.go` implementations